src/vehicle.h

```diff
@@ -1109,8 +1110,8 @@ class vehicle
         // Get maximum velocity for the current movement mode
         int safe_velocity( bool fueled = true ) const;
 
-        // Generate smoke from a part, either at front or back of vehicle depending on velocity.
-        void spew_smoke( double joules, int part, int density = 1 );
+        // Generate field from a part, either at front or back of vehicle depending on velocity.
+        void spew_field( double joules, int part, field_type_id type = fd_null, int density = 1 );
```

Comment:
The `type` is optional and defaults to `fd_null`. When would I leave out the type? Does it even make sense to spawn a `fd_null` field? Is that even possible?
```suggestion
        void spew_field( double joules, int part, field_type_id type, int density = 1 );
```